gwcs/tests/test_spectroscopy_models.py

```diff
@@ -47,3 +48,42 @@ def test_wavelength_grating_equation_units():
 
     result = model(-u.Quantity(alpha_in), -u.Quantity(alpha_in))
     assert_allclose(result, wave)
+
+
+@pytest.mark.parametrize(('wavelength', 'n'),
+                         [(1, 1.43079543),
+                          (2, 1.42575377),
+                          (5, 1.40061966)
+                          ])
+def test_SellmeierGlass(wavelength, n):
+    """ Test from Nirspec team.
+
+    Wavelength is in microns.
+    """
+
+    B_coef =  [0.58339748, 0.46085267, 3.8915394]
+    C_coef = [0.00252643, 0.010078333, 1200.556]
+    model = sp.SellmeierGlass(B_coef, C_coef)
+    n_result = model(wavelength)
+    assert_allclose(n_result, n)
+
+
+def test_SellmeierZemax():
+    """ The data for this test come from Nirspec."""
+    kcoef = [0.58339748, 0.46085267, 3.8915394]
+    lcoef = [0.00252643, 0.010078333, 1200.556]
+    D_coef = [-2.66e-05, 0.0, 0.0]
+    E_coef = [0., 0., 0.]
+    model = sp.SellmeierZemax(65, 35, 0, 0, B_coef = kcoef,
+                              C_coef=lcoef, D_coef=D_coef,
+                              E_coef=E_coef)
+    n = 1.4254647475849418
+    assert_allclose(model(2), n)
+
+
+def test_Snell3D():
+    """ Test from Nirspec."""
+    n = 1.4254647475849418
+    model = sp.Snell3D(n)
+    expected = (0.07015255913513296, 0.07015255913513296, 0.9950664484814988)
+    assert_allclose(model(.1, .1, .1), expected)
```

Comment:
Though the input direction cosines are impossible (right?), but it doesn't matter since the z component is never used.

Reply:
Yah, I should fix it. I just wanted some value to compare with.